kgforge/core/resource.py

```diff
@@ -95,6 +95,36 @@ def _set_synchronized(self, sync: bool) -> None:
     
     _synchronized = property(_get_synchronized, _set_synchronized)
 
+    def has_identifier(self, return_attribute: bool = False) ->  Union[bool, Tuple[bool, str]]:
```

Comment:
```suggestion
    def has_term(self, term, return_attribute: bool = False) ->  Union[bool, Tuple[bool, str]]:
            result = False
            attribute = None
            at_term = f"@{term}"
            if hasattr(self, at_term):
                result : True
                attribute = at_term
            if hasattr(self, term):
                result: True
                attribute = term
            return result, attribute if return_attribute else result
```

Reply:
To replace has_type or has_identifier, and have a single method for any term. Of course, one would need to update the other methods: the `get_type` method will pass `term=type` and `get_identifier` `term=id`.